pkg/istiovalues/fips.go

```diff
@@ -0,0 +1,46 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istiovalues
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/istio-ecosystem/sail-operator/pkg/helm"
+)
+
+const (
+	FipsEnableFilePath = "/proc/sys/crypto/fips_enabled"
+	TestModeFilePath   = "/tmp/fips_enabled"
+)
+
+// ApplyFipsValues: Detect FIPS enabled mode on OpenShift and set value pilot.env.COMPLIANCE_POLICY
+// To avoid changing the host machine FIPS_ENABLED_FILE_PATH configuration,
+// set testMode to true when running unit tests.
+func ApplyFipsValues(mergedHelmValues helm.Values, testMode bool) (helm.Values, error) {
+	fipsEnabledFile := FipsEnableFilePath
+	if testMode {
+		fipsEnabledFile = TestModeFilePath
+	}
+
+	contents, err := os.ReadFile(fipsEnabledFile)
+	if err == nil && string(contents) == "1\n" {
+		fmt.Println("Running on FIPS enabled system.")
+		if err = mergedHelmValues.SetIfAbsent("pilot.env.COMPLIANCE_POLICY", "fips-140-2"); err != nil {
```

Comment:
Does the `"fips-140-2"` need to be configurable (e.g. via an environment variable for the operator)?

Reply:
I don't think that need to be exposed at the operator layer. According to the upstream note[1]. It's an environment variable for the istiod container. So our operator pod doesn't need that env.

I haven't heard a use case to set this value or other policy value using sail operator. If we need to add a field from our CRD, we can add it as a feature in later release. What do you think ?

[1] https://github.com/openshift-service-mesh/istio/blob/master/releasenotes/notes/fips.yaml